### ISSUE TITLE
[BUG] Not able to connect to solr server with username and password for authentication

### ISSUE THREAD

**Issue by JenHa** [TYPO3-Solr/ext-solr#2540]
**Describe the bug**
Till version 9.x where the configuration was made over typoscript it was possible to configure username and password along with scheme, host, path etc. This option isn't available in the sites config module anymore. I tried to confiugre username and password over typoscript, but still the connection couldn't be made. Removing all configuration from the site config leads to using the default config (localhost etc.) instead of the made typoscript configuration. 
https://github.com/TYPO3-Solr/ext-solr/blob/master/Configuration/TypoScript/Solr/setup.txt#L17 here the options are still in the typoscript, so I thought they could be used, but doesn't seem like they are used anywhere. 

**To Reproduce**
1. Setup TYPO3 9LTS and install ext-solr in version 10.x
2. Setup a solr server with basic auth to prevent unwanted access. 
3. Configure solr over site module (all options are correctly used, but username and password can't be configured) or typoscript (default configuration will be used)
4. Go to the solr info module. It will stat, that no connection could be possible

**Expected behavior**
Username and password can either be configured over the sites module or the typoscript options will be used as a backup option. In both cases a connection to a solr server with basic auth should be possible. 

**Used versions:**
 - TYPO3 Version: 9.5.14
 - EXT:solr Version: 10.0.1
 - Used Apache Solr Version: 8.2.0

**Comment by JenHa** [comment]
Update: I debugged a little. The auth information is correctly given to the solarium guzzle class (solarium/core/client/adapter/), if it's written in the config.yaml . The request contains the needed auth information. But when I tried to debug the response with xdebug, it didn't get to the point, after the request, where the response should have been progressed, but instead directly jumped to the part where the connection is written in the missinghosts-array (https://github.com/TYPO3-Solr/ext-solr/blob/master/Classes/Controller/Backend/Search/InfoModuleController.php#L141). Maybe it's not waiting for the asynchron request to resolve (the method returns a promise). 

This could explain why neither using sorl_username_read/solr_password_read nor writing the auth into the host (like username:password@host) worked. 
Anybody else having this problem or can help me?

Thanks in advance
